RockWeb/Blocks/Connection/ConnectionOpportunitySignup.ascx.cs

```diff
@@ -357,6 +357,14 @@ public void ShowDetail( int opportunityId )
                     ShowError( "Inactive Opportunity Type", "The opportunity is not currently active." );
                     return;
                 }
+                
+                var opportunityConnectionTypeId = opportunity.ConnectionTypeId;
```

Comment:
I like what your on to...

There are some small formatting issues with the code below, but you can actually access the opportunity's `ConnectionType` directly since it was fully loaded in the `Get()` method above.  So you could actually just amend line (roughly) 354 and do something like this:

```CSharp
                if ( !opportunity.IsActive || ! opportunity.ConnectionType.IsActive )
                {
                    pnlSignup.Visible = false;
                    ShowError( "Inactive Opportunity Type", "The opportunity is not currently active." );
                    return;
                }
```

Additionally, you might want to follow this through and make similar adjustments to the other externally facing blocks... possibly Connection Opportunity Search (OpportunitySearch.ascx.cs) and Connection Opportunity Detail Lava as they may have the same issue.

Reply:
I did add a pr for Connection Opportunity Search as I found that earlier as well. I wasn't aware of the way that you have shown and yes that would make more since. I haven't looked at the Detail Lave block yet.

First time and all, so should I squash these pr's, refactor and then submit?

Reply:
@TheArkChurch Since it's your first PR here, I'll let the squash slide after you submit the correction commit, as long as you DM me with your actual name too :)

Reply:
Changes tested and applied

Reply:
And yes this issue is present in Connection Opportunity Detail Lava as well but I have yet to find how to fix that one yet.